plugins/modules/lightsail.py

```diff
@@ -316,6 +354,22 @@ def main():
         user_data=dict(type='str', default=''),
         wait=dict(type='bool', default=True),
         wait_timeout=dict(default=300, type='int'),
+        public_ports_info=dict(type='list', elements='dict', default=[
+            {
+                'fromPort': 22,
+                'toPort': 22,
+                'protocol': 'tcp',
+                'cidrs': ['0.0.0.0/0'],
+                'ipv6Cidrs': ['::/0'],
+            },
+            {
+                'fromPort': 80,
+                'toPort': 80,
+                'protocol': 'tcp',
+                'cidrs': ['0.0.0.0/0'],
+                'ipv6Cidrs': ['::/0'],
+            },
+        ])
```

Comment:
```suggestion
        public_ports=dict(type='list', elements='dict')
```

By passing a default here, there's the risk that you'll overwrite something that was explicitly configured by the user (especially with a new parameter).

I'd recommend *not* passing a default.


Additionally, these parameters should generally be snake_case rather than CamelCase, and it would be good to verify the "shape" before passing it to the SDK/API.  This makes it easier for us to return consistent error messages.

https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec

specifically adding `options` to the public_ports information